cinema/views.py

```diff
@@ -1,45 +1,120 @@
-from rest_framework.decorators import api_view
 from rest_framework.response import Response
 from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.viewsets import GenericViewSet, ModelViewSet
+from rest_framework.request import Request
+from rest_framework.generics import GenericAPIView
+from rest_framework.mixins import (
+    ListModelMixin,
+    CreateModelMixin,
+    UpdateModelMixin,
+    RetrieveModelMixin,
+    DestroyModelMixin,
+)
 
 from django.shortcuts import get_object_or_404
 
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
+from cinema.models import Movie, Genre, Actor, CinemaHall
+from cinema.serializers import (
+    MovieSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    CinemaHallSerializer,
+)
 
 
-@api_view(["GET", "POST"])
-def movie_list(request):
-    if request.method == "GET":
-        movies = Movie.objects.all()
-        serializer = MovieSerializer(movies, many=True)
+class GenreList(APIView):
+    def get(self, request: Request) -> Response:
+        genres = Genre.objects.all()
+        serializer = GenreSerializer(genres, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "POST":
-        serializer = MovieSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
-
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    def post(self, request: Request) -> Response:
+        serialized_data = GenreSerializer(data=request.data)
+        if serialized_data.is_valid():
+            serialized_data.save()
+            return Response(
+                serialized_data.data, status=status.HTTP_201_CREATED
+            )
+        return Response(
+            serialized_data.errors, status=status.HTTP_400_BAD_REQUEST
+        )
 
```

Comment:
in all places like this, you can refactor following the DRF documentation using
serializer.is_valid(raise_exception=True)
https://www.django-rest-framework.org/api-guide/serializers/#raising-an-exception-on-invalid-data:~:text=Raising%20an%20exception%20on%20invalid%20data